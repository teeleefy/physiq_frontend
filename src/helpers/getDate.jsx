function getDate() {
    const today = new Date();
    const year = today.getFullYear();
    let month = today.getMonth() + 1;
    if(month<10){
        month = `0${month}`
    }
    let date = today.getDate();
    if(date<10){
        date = `0${date}`
    }
    return `${year}-${month}-${date}`;
  }

export default getDate;
