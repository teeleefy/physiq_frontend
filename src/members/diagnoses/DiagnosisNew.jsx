import { Form, FormGroup, Input, Button, Label, Alert } from "reactstrap";
import { useState, useContext } from "react";
import {MemberContext} from "../../auth/UserContext.js";
import PhysiqApi from "../../Api.js";
import { useNavigate } from "react-router-dom";
import '../styles/Diagnosis.css'
import { toast } from 'react-toastify';

function DiagnosisNew({getDate}){
    let navigate = useNavigate();
    const { currentMember } = useContext(MemberContext);
    const [formMessages, setFormMessages] = useState([]);
    const [updateSuccess, setUpdateSuccess] =useState();
    const [formData, setFormData] = useState({name: "", dateReceived: "", notes: ""});
    const todaysDate = getDate();

    async function addDiagnosis() {
        try {
          let newDiagnosis= await PhysiqApi.addMemberDiagnosis(currentMember.id, formData);
          return { success: true };
        } catch (errors) {
          console.error("Add Diagnosis failed", errors);
          return { success: false, errors };
        }
      }

    async function handleSubmit(evt){
        evt.preventDefault();
        let result = await addDiagnosis();
        
        if(result.success){
            toast.success('Diagnosis Added!', {
                position: "top-center",
                autoClose: 2000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: true,
                draggable: true,
                progress: undefined,
                theme: "dark",
                }); 
           setFormMessages(['Diagnosis Added!'])
           setUpdateSuccess(true);
           navigate("..", { relative: "path"});
        }
        else{
            setFormMessages(result.errors);
            setUpdateSuccess(false);
        }
    };
  
    /** Update local state w/curr state of input elem */
  
    const handleChange = evt => {
      const { name, value }= evt.target;
      setFormData(fData => ({
        ...fData,
        [name]: value
      }));
    };

    return(
        <>
            <Form className="DiagnosisNew-Form m-4" >
            <h1 className="Diagnosis-h1">Add Diagnosis:</h1>
            <p className="text-secondary">Fields marked with <span className="text-danger">*</span> are required.</p>
            <FormGroup>
                <Label className="Diagnosis-label" for="name" >
                    <b>Diagnosis Name</b><span className="text-danger">*</span>
                </Label>
                <Input
                id="name"
                name="name"
                maxLength={125}
                value={formData.name}
                placeholder="Enter diagnosis name..."
                type="text"
                onChange={handleChange}
                />
            </FormGroup>
            <FormGroup className="Diagnosis-date-input">
                <Label className="Diagnosis-label" for="dateReceived">
                    <b>Date Received</b>
                </Label>
                <Input
                id="dateReceived"
                name="dateReceived"
                value={formData.dateReceived}
                max={todaysDate}
                placeholder="Enter date received..."
                type="date"
                onChange={handleChange}
                />
                </FormGroup>
            <FormGroup>
                <Label className="Diagnosis-label" for="notes">
                    <b>Notes</b>
                </Label>
                <Input
                id="notes"
                name="notes"
                maxLength={250}
                value={formData.notes}
                placeholder="Enter notes..."
                type="textarea"
                onChange={handleChange}
                />
            </FormGroup>
            <p className="text-secondary">{formData.notes.length}/250 Characters</p>
                {formMessages.length
                    ? formMessages.map(msg => <Alert color={updateSuccess ? "success": "danger"}>{msg}</Alert>)
                    : null
                }
                
                <Button className="btn-dark" onClick={handleSubmit}>Add Diagnosis</Button>
            </Form>
        </>
    )}
    
export default DiagnosisNew;