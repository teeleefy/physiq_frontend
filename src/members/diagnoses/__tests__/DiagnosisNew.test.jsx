import {renderWithRouterAndMemberContext, screen} from '../../../../__tests__/test-tools';
import getDate from "../../../helpers/getDate.jsx"
import DiagnosisNew from '../DiagnosisNew';

describe('DiagnosisNew', () => {
    it('renders the DiagnosisNew component', () => {
      renderWithRouterAndMemberContext(<DiagnosisNew getDate={getDate}/>);
      screen.debug(); 
    });

    it("matches snapshot", function() {
      const { asFragment } = renderWithRouterAndMemberContext(<DiagnosisNew getDate={getDate}/>);
      expect(asFragment()).toMatchSnapshot();
    });

  });
