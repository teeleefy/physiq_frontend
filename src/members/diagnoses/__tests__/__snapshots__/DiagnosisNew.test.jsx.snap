// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DiagnosisNew > matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="DiagnosisNew-Form m-4"
  >
    <h1
      class="Diagnosis-h1"
    >
      Add Diagnosis:
    </h1>
    <p
      class="text-secondary"
    >
      Fields marked with 
      <span
        class="text-danger"
      >
        *
      </span>
       are required.
    </p>
    <div
      class="mb-3"
    >
      <label
        class="Diagnosis-label form-label"
        for="name"
      >
        <b>
          Diagnosis Name
        </b>
        <span
          class="text-danger"
        >
          *
        </span>
      </label>
      <input
        class="form-control"
        id="name"
        maxlength="125"
        name="name"
        placeholder="Enter diagnosis name..."
        type="text"
        value=""
      />
    </div>
    <div
      class="Diagnosis-date-input mb-3"
    >
      <label
        class="Diagnosis-label form-label"
        for="dateReceived"
      >
        <b>
          Date Received
        </b>
      </label>
      <input
        class="form-control"
        id="dateReceived"
        max="2024-08-06"
        name="dateReceived"
        placeholder="Enter date received..."
        type="date"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="Diagnosis-label form-label"
        for="notes"
      >
        <b>
          Notes
        </b>
      </label>
      <textarea
        class="form-control"
        id="notes"
        maxlength="250"
        name="notes"
        placeholder="Enter notes..."
      />
    </div>
    <p
      class="text-secondary"
    >
      0/250 Characters
    </p>
    <button
      class="btn-dark btn btn-secondary"
      type="button"
    >
      Add Diagnosis
    </button>
  </form>
</DocumentFragment>
`;
